codalab/model/bundle_model.py

```diff
@@ -677,6 +677,85 @@ def get_bundle_uuids(self, conditions, max_results):
 
         return self._execute_query(query)
 
+    def get_memoized_bundles(self, user_id, command, dependencies):
+        '''
+        Get the most recently finished bundle that matches with input command and dependencies.
+        :param user_id: a string that specifies the current user id.
+        :param command: a string that defines the command that is used to search for memoized bundles in the database.
+        :param dependencies: a list of string in the structure of ["key1:uuid1","key2:uuid2"] defines
+                            dependencies that are used to search for memoized bundles in the database.
+        :return: a list of matched uuids.
+        '''
+        # Get a list of uuids that matches with the given dependencies from the cl_bundle table.
+        # This step is to standardize uuid for future comparison.
+        with self.engine.begin() as connection:
+            # Select bundles that has the given command from the bundle table.
```

Comment:
You should ideally do a single DB query that has the command + dependencies for efficiency.

Reply:
My intention of `filter_on_command` first is to reduce the size of the records that will be searching on before the later join/selection operation. But it might not be significant in improving efficiency comparing to executing with one query.